plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/createFromUsage/createCallable/CreateCallableFromCallActionFactory.kt

```diff
@@ -96,24 +97,32 @@ sealed class CreateCallableFromCallActionFactory<E : KtExpression>(
                 if (containers.isNotEmpty()) containers else return null
             } else listOf()
 
-        return doCreateCallableInfo(element, analysisResult, calleeExpr.getReferencedName(), receiverType, possibleContainers)
+        return doCreateCallableInfo(
+            element,
+            analysisResult,
+            calleeExpr.getReferencedName(),
+            receiverType,
+            isForCompanion,
+            possibleContainers
+        )
     }
 
-    private fun getReceiverTypeInfo(context: BindingContext, project: Project, receiver: Receiver?): TypeInfo? {
+    private fun getReceiverTypeInfo(context: BindingContext, project: Project, receiver: Receiver?): Pair<TypeInfo, Boolean>? {
         return when (receiver) {
-            null -> TypeInfo.Empty
+            null -> TypeInfo.Empty to false
             is Qualifier -> {
                 val qualifierType = context.getType(receiver.expression)
-                if (qualifierType != null) return TypeInfo(qualifierType, Variance.IN_VARIANCE)
+                if (qualifierType != null) return TypeInfo(qualifierType, Variance.IN_VARIANCE) to false
 
                 if (receiver !is ClassQualifier) return null
                 val classifierType = receiver.descriptor.classValueType
-                if (classifierType != null) return TypeInfo(classifierType, Variance.IN_VARIANCE)
+                if (classifierType != null) return TypeInfo(classifierType, Variance.IN_VARIANCE) to false
 
-                val javaClassifier = receiver.descriptor as? JavaClassDescriptor ?: return null
+                val javaClassifier = receiver.descriptor as? JavaClassDescriptor
+                    ?: return TypeInfo(receiver.descriptor.defaultType, Variance.IN_VARIANCE) to true
```

Comment:
I just found `TypeInfo.StaticContextRequired`. It seems that the boolean flag `isForCompanion` you introduced is actually unnecessary and can be replaced with `org.jetbrains.kotlin.idea.quickfix.createFromUsage.callableBuilder.TypeInfo#getStaticContextRequired`